### DISCUSSION TITLE
Problem with XMLSignatureValidator

### DISCUSSION THREAD

**Discussion by lgignoud** [airsdk/Adobe-Runtime-Support#2618] [Q&A]
Hi,
We have a Flex application using Air 32 on Windows platform. We would happily switch to Harman runtime because of the great new features on the Windows desktop (WebView, multiple instances, etc.). So I gave it a try with our application.
Unfortunately we use the flash XML signature validation, and it seems to have a problem with the new Air runtime. 
Our code is very similar to the one in this example:
https://help.adobe.com/fr_FR/as3/dev/WSe3d2d52902616553-16c3561611c42fb32b6-8000.html
We use the latest runtime: 50.2.2. After the code reaches:
`verifier.verify( xmlSignature ); `
nothing happens, and neither Event.COMPLETE or ErrorEvent.ERROR are triggered.

Do you have an idea of the problem ?

**Comment by lgignoud** [comment]
No luck with our issue... 
@ajwfrost Do you need more data to investigate ? Except for this problem, our app seems to run fine with the 50.2.2 runtime. Feel free to reach me if we need to perform additionnal steps to go further.

**Reply by ajwfrost** [reply]
Hi - sorry, I thought I'd already responded on this thread! We had implemented what (we thought) was needed for the XML signature validation, and it's working in terms of verifying the signatures.xml file that's created within an AIR package. But if you're able to package up your application (or a test case) using the ADT "bundle" command, and zip this up and send it to us, we can try to find out more about what's happening..

thanks

**Reply by lgignoud** [reply]
Thanks Andrew, I will try to provide you an example !

**Reply by lgignoud** [reply]
Hi Andrew,
Please find attached a simple IntelliJ project, where I validate a file on the desktop called "signatures.xml". It's the file you find on a any Air distribution, under META-INF. When compiled with Air32, no problem. When compiled with Air50.2, no event of the validator is called back. I use also Flex 4.16.1, if it is of any help.
Thanks in advance for your feedback.

[XmlSignatureTest.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/11407346/XmlSignatureTest.zip)

**Reply by ajwfrost** [reply]
Great, thanks -> so the issue in our versions of AIR is that we had enabled the synchronous verification process, but this uses an asynchronous verification process that needs more configuring/building... background being that there are some parts of the Adobe version of AIR that we don't have access to, around DRM (which this capability was linked with). So we will need to check to see if we can configure things to work asynchronously, hopefully using some form of the existing code, or if not to create it ourselves..

... and from the looks of things, actually it may be fairly straightforward. Or at least, we get the "complete" event - now just trying to find out which status value should actually be set to what based on the input file.. we have a validly signed file showing "unknown" for the identity, so maybe there's more work to do here...

**Reply by lgignoud** [reply]
Thanks for this rapid feedback. In fact we don't really need asynchronous validation, I wasn't aware that you can do it synchronously. How can I try the synchronous process ?

**Reply by ajwfrost** [reply]
Hi - actually upon inspection, it looks like the synchronous version is only used internally i.e. checking the signatures at start-up etc.

We can get the asynch mechanism set up in the next release, the problem above where a validly signed file wasn't getting the identity was just because we'd forgotten to set the `useSystemTrustStore` property to 'true' - which is needed if the signing certificate is validated by a root certificate available on the OS (which should generally be the case now, I believe!)

thanks

**Reply by lgignoud** [reply]
Thanks andrew, can't wait to test it with the next release !
